binary_search_tree/tree.py

```diff
@@ -1,59 +1,148 @@
 class TreeNode:
-    def __init__(self, key, val = None):
-        if val == None:
-            val = key
+    def __init__(self, key, value = None):
+        if value == None:
+            value = key
 
         self.key = key
-        self.value = val
+        self.value = value
         self.left = None
         self.right = None
         
-
-
 class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log n) if balanced
+    # Space Complexity: O(1)
     def add(self, key, value = None):
-        pass
+        if self.root == None:
+            self.root = TreeNode(key, value)
+        else:
+            self.add_helper(self.root, key, value)
+    
+    def add_helper(self, current, key, value):
+        if current == None:
+            return TreeNode(key, value)
+        if current.key > key:
+            current.left = self.add_helper(current.left, key, value)
+        else:
+            current.right = self.add_helper(current.right, key, value)
+        return current 
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log n) if balanced
+    # Space Complexity: O(1)
     def find(self, key):
```

Comment:
👍 However you made a recursive solution which means your space complexity is O(log n).  This is due to the recursive call stack.